modelscope/models/cv/image_depth_estimation_marigold/networks/marigold_model.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2023 Bingxin Ke, ETH Zurich. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# --------------------------------------------------------------------------
+# If you find this code useful, we kindly ask you to cite our paper in your work.
+# Please find bibtex at: https://github.com/prs-eth/Marigold#-citation
+# More information about the method can be found at https://marigoldmonodepth.github.io
+# --------------------------------------------------------------------------
+
+
+from typing import Dict, Union
+
+import torch
+from torch.utils.data import DataLoader, TensorDataset
+import numpy as np
+from tqdm.auto import tqdm
+from PIL import Image
+
+from diffusers import (
+    DiffusionPipeline,
+    DDIMScheduler,
+    UNet2DConditionModel,
+    AutoencoderKL,
+)
+from diffusers.utils import BaseOutput
+from transformers import CLIPTextModel, CLIPTokenizer
+
+from .util.image_util import chw2hwc, colorize_depth_maps, resize_max_res
+from .util.batchsize import find_batch_size
+from .util.ensemble import ensemble_depths
+
+
+class MarigoldDepthOutput(BaseOutput):
+    """
+    Output class for Marigold monocular depth prediction pipeline.
+
+    Args:
+        depth_np (`np.ndarray`):
+            Predicted depth map, with depth values in the range of [0, 1].
+        depth_colored (`PIL.Image.Image`):
+            Colorized depth map, with the shape of [3, H, W] and values in [0, 1].
+        uncertainty (`None` or `np.ndarray`):
+            Uncalibrated uncertainty(MAD, median absolute deviation) coming from ensembling.
+    """
+
+    depth_np: np.ndarray
+    depth_colored: Image.Image
+    uncertainty: Union[None, np.ndarray]
+
+
+class MarigoldPipeline(DiffusionPipeline):
```

Comment:
this pipeline should be moved to image_depth_estimation_pipeline

Reply:
this is the Marigold model which is inherited from the DiffusionPipeline. Is it a solution that I change the name to MarigoldModel?